### PR TITLE
Java: Model `io.jsonwebtoken.SigningKeyResolverAdapter` and `io.jsonwebtoken.JwsHeader`

### PR DESCRIPTION
**Pull request by egregius313**
This adds MaD models to represent APIs from the `io.jsonwebtoken` library.

### CODE REVIEW

File: java/ql/lib/ext/io.jsonwebtoken.model.yml
Comment (atorralba) [inline java/ql/lib/ext/io.jsonwebtoken.model.yml:9]: We're risking having spurious flow when tainting the header through the algorithm and getting the key ID, or vice-versa, e.g.

```java
header.setAlgorithm(tainted());
sink(header.getKeyId());
```

To solve that, we'd need to use synthetic fields:

```suggestion
      - ["io.jsonwebtoken", "JwsHeader", True, "getAlgorithm", "", "", "Argument[this].SyntheticField[io.jsonwebtoken.JwsHeader.algorithm]", "ReturnValue", "taint", "manual"]
      - ["io.jsonwebtoken", "JwsHeader", True, "setAlgorithm", "", "", "Argument[0]", "Argument[this].SyntheticField[io.jsonwebtoken.JwsHeader.algorithm]", "taint", "manual"]
      - ["io.jsonwebtoken", "JwsHeader", True, "getKeyId", "", "", "Argument[this].SyntheticField[io.jsonwebtoken.JwsHeader.keyId]", "ReturnValue", "taint", "manual"]
      - ["io.jsonwebtoken", "JwsHeader", True, "setKeyId", "", "", "Argument[0]", "Argument[this].SyntheticField[io.jsonwebtoken.JwsHeader.keyId]", "taint", "manual"]
```

But then you'll need a `TaintInheritingContent` to establish that a generally-tainted `JwsHeader` (like one that is a source) also taints its synthetic fields, like the one we have for `Intent.extras`:

https://github.com/github/codeql/blob/feb31612f5c4bfece15d233edcebc2069c3b50b5/java/ql/lib/semmle/code/java/frameworks/android/Intent.qll#L125-L129

I'll leave the final call to you. If you don't observe FPs caused by this spurious flow in real world applications, we can leave it as it is if you want.

**Review by atorralba — COMMENTED** [review]
Mostly LGTM, with one comment.

File: java/ql/lib/ext/generated/io.jsonwebtoken.model.yml
Comment (jcogs33) [inline java/ql/lib/ext/generated/io.jsonwebtoken.model.yml:1]: Were the models in this file reviewed for correctness at all, and if so, what was the process for validating them? 
Asking because I want to make sure I understand expectations regarding validating df-generated models before merging them.

File: java/ql/lib/ext/generated/io.jsonwebtoken.model.yml
Comment (jcogs33) [inline java/ql/lib/ext/generated/io.jsonwebtoken.model.yml:304]: I didn't realize from the discussion elsewhere that these models are from `java.util`, not `io.jsonwebtoken`. It seems potentially odd that the model generator is including these when creating models for the `io.jsonwebtoken` framework. I wonder if it's worth looking into this discrepancy further?

Comment (aschackmull) [reply java/ql/lib/ext/generated/io.jsonwebtoken.model.yml:304]: These should definitely not be added.

File: java/ql/lib/semmle/code/java/frameworks/IoJsonWebToken.qll
Comment (aschackmull) [inline java/ql/lib/semmle/code/java/frameworks/IoJsonWebToken.qll:10]: I don't expect this to work, if this file is not bidirectionally imported.

Comment (egregius313) [reply java/ql/lib/semmle/code/java/frameworks/IoJsonWebToken.qll:10]: Correct, it had caused some problems. I added an import to `FlowSteps::Frameworks` and it seems to work now.

**Review by atorralba — APPROVED** [review]

**Pull request merged**
